cql3/statements/ks_prop_defs.cc

```diff
@@ -139,28 +139,22 @@ data_dictionary::storage_options ks_prop_defs::get_storage_options() const {
     return opts;
 }
 
-ks_prop_defs::init_tablets_options ks_prop_defs::get_initial_tablets(const sstring& strategy_class, bool enabled_by_default) const {
-    // FIXME -- this should be ignored somehow else
-    init_tablets_options ret{ .enabled = false, .specified_count = std::nullopt };
-    if (locator::abstract_replication_strategy::to_qualified_class_name(strategy_class) != "org.apache.cassandra.locator.NetworkTopologyStrategy") {
-        return ret;
-    }
-
+std::optional<unsigned> ks_prop_defs::get_initial_tablets(std::optional<unsigned> default_value) const {
     auto tablets_options = get_map(KW_TABLETS);
     if (!tablets_options) {
-        return enabled_by_default ? init_tablets_options{ .enabled = true } : ret;
+        return default_value;
```

Comment:
This `default_value` is used only in one place, maybe it'd increase the readability if you skipped this function parameter and just reutrned `{}`? Also, at least to me, `default_value` reads like something else than a disengaged optional and I'd expect something else, e.g. `0` (but it's only me).

Reply:
I cannot return disengaged optional here, as it's also returned lower in tablets = { enabled: false } branch, thus meaning that tablets should be disabled